substrate/frame/support/src/lib.rs

```diff
@@ -779,6 +781,32 @@ macro_rules! assert_error_encoded_size {
 	} => {};
 }
 
+/// Do something hypothetically by rolling back any changes afterwards.
+///
+/// Returns the original result of the closure.
+#[macro_export]
+macro_rules! experimental_hypothetically {
+	( $e:expr ) => {
+		$crate::storage::transactional::with_transaction(
+							|| -> $crate::__private::TransactionOutcome<Result<_, $crate::__private::DispatchError>> {
+								$crate::__private::TransactionOutcome::Rollback(Ok($e))
+							},
+						)
+						.expect("Always returning Ok; qed")
+	};
+}
+
+/// Assert something to be *hypothetically* `Ok` without actually committing it.
+///
+/// Reverts any storage changes made by the closure.
+#[macro_export]
+macro_rules! experimental_hypothetically_ok {
+	($e:expr $(, $args:expr)* $(,)?) => {
+		let result = $crate::experimental_hypothetically!($e);
+		$crate::assert_ok!(result $(, $args)*);
+	};
+}
```

Comment:
can we use the experimental feature flag with `macro_rules!`?

Reply:
Looks like this is possible 🤔 I did not know.